src/api/app/models/project.rb

```diff
@@ -586,6 +586,9 @@ def update_from_xml(xmlhash, force = nil)
 
   def update_repositories(xmlhash, force)
     fill_repo_cache
+    # Some repositories might be refered by path elements before they appear in th
```

Comment:
I would add something that this is the situation when your create multiple repositories in a project where one repo uses another one.